aio_pika/robust_connection.py

```diff
@@ -134,9 +134,18 @@ def channel(self, channel_number: int = None,
         for number, channel in self._channels.items():
             try:
                 await channel.on_reconnect(self, number)
-            except CONNECTION_EXCEPTIONS:
+            except CONNECTION_EXCEPTIONS as e:
                 log.exception('Open channel failure')
-                await self.close()
+
+                if self.connection is not None:
+                    await asyncio.gather(
```

Comment:
Why `gather` but not just `self.close(e)`?

Reply:
`self.close()` will set `self.__closed` flag. And connection couldn't be reopened after.

Reply:
Ah, understand.